### ISSUE TITLE
"Cannot fork" error when building with 3.1.39 (coming from 2.0.34)

### ISSUE THREAD

**Issue by aduermael** [emscripten-core/emscripten#19409]
We're trying to update the emsdk to build our application. 
We were still using `2.0.34`.

We installed `3.1.39` and could build our app without issues. But at runtime, we're getting these logs in the console: 
```
Cannot fork. Function not implemented [52]
FAILED: 279 of 279 unit tests have failed.
```

It seems to point to that line specifically: 

<img width="394" alt="Screenshot 2023-05-21 at 21 35 36" src="https://github.com/emscripten-core/emsdk/assets/6775074/7190b000-7e13-4399-b688-4b7d1ca507f8">

We've been trying to understand that, looked around for similar issues, but we're unfortunately still clueless. 

Tried to `emcc --clear-cache` + flushing build directory.

I don't think `fork()` used to be allowed either with `2.0.34`, and it still works for us when sticking to that version.

**Comment by sbc100** [comment]
The fork system call certainly never worked in emscripten, and I don't think it ever returned success, even with 2.0.34. 

I would imagine that calling fork() would always result in a `ENOSYS` being returned.  It certainly does that today: https://github.com/emscripten-core/emscripten/blob/98b618f09bc106d401c0045fa7252f1cd15bd3c0/system/lib/libc/emscripten_libc_stubs.c#L169

It looks like that stub was created in #15209 which essentially moved the stub from JS code to native code. And that PR was released in 2.0.32.

My advice would be to try to track down where your code is calling `fork`, and why?  And was it somehow not calling `fork()` in the working build?  Or was it somehow ignoring or tolerating the error return code from `fork()`

**Comment by aduermael** [comment]
Ok, we found the problem! 
With the new SDK, the wrong `main` function was executed. 
Made us realize we were compiling & linking useless test code... 
We were actually lucky that it worked with emsdk 2.x.

Thanks for your help anyway @sbc100!
